src/ring_theory/algebra.lean

```diff
@@ -185,6 +186,12 @@ instance of_subsemiring (S : subsemiring R) : algebra S A :=
   smul_def' := λ r x, algebra.smul_def r x,
   .. (algebra_map R A).comp (subsemiring.subtype S) }
 
+/-- A semiring that is an algebra over a commutative ring carries a natural ring structure. -/
+def as_ring (R A : Type*) [comm_ring R] [semiring A] [algebra R A] : ring A :=
+{ neg          := λ a, (-1 : R) • a,
+  add_left_neg := λ a, by { nth_rewrite 1 ← one_smul _ a, rw [← add_smul, add_left_neg, zero_smul], },
+  ..(infer_instance : semiring A), }
```

Comment:
I think there's a similar proof in the `exterior_algebra` branch - clearly we need this in mathlib soon!

https://github.com/leanprover-community/mathlib/compare/exterior_algebra#diff-ba0f67e086711a721e580c048767b8d5R306-R317

Reply:
Although your proof looks perhaps a little nicer

Reply:
They're basically the same I think!

Reply:
I don't think `exterior_algebra` is landing any time soon - my point is that if your branch doesn't either, it would make sense to put `as_ring` in a standalone PR that everyone can benefit from sooner.

Reply:
Ah I see, that's a good point.

Please feel free to create a PR inserting `ring_of_comm_ring_algebra` (which I think is a better name) if you wish.

Reply:
Done in #4252